interface/modules/custom_modules/oe-module-prior-authorizations/ModuleManagerListener.php

```diff
@@ -0,0 +1,136 @@
+     */
+    public function moduleManagerAction($methodName, $modId, string $currentActionStatus = 'Success'): string
+    {
+        if (method_exists(self::class, $methodName)) {
+            return self::$methodName($modId, $currentActionStatus);
+        } else {
+            // no reason to report, action method is missing.
+            return $currentActionStatus;
+        }
+    }
+
+    /**
+     * Required method to return namespace
+     * If namespace isn't provided return empty string
+     * and register namespace at top of this script..
+     *
+     * @return string
+     */
+    public static function getModuleNamespace(): string
+    {
+        // Module Manager will register this namespace.
+        return 'Juggernaut\\OpenEMR\\Modules\\PriorAuthModule\\';
+    }
+
+    /**
+     * Required method to return this class object
+     * so it will be instantiated in Laminas Manager.
+     *
+     * @return ModuleManagerListener
+     */
+    public static function initListenerSelf(): ModuleManagerListener
+    {
+        return new self();
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+    private function help_requested($modId, $currentActionStatus): mixed
+    {
+        // must call a script that implements a dialog to show help.
+        // I can't find a way to override the Lamina's UI except using a dialog.
+        if (file_exists(__DIR__ . '/show_help.php')) {
+            include __DIR__ . '/show_help.php';
+        }
+        return $currentActionStatus;
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+    private function enable($modId, $currentActionStatus): mixed
+    {
+        // Return the current action status from Module Manager in case of error from its action.
+        return $currentActionStatus;
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+    private function disable($modId, $currentActionStatus): mixed
+    {
+        return $currentActionStatus;
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+    private function unregister($modId, $currentActionStatus)
+    {
+        $logMessage = 'Prior Auth table have been removed if empty else manually remove'; // Initialize an empty string to store log messages
+        $records = sqlQuery("SELECT * FROM `module_prior_authorizations` ORDER BY id DESC LIMIT 1"); // Query the database for all records in the module_prior_authorizations table
+        if (empty($records)) { // Check if the records array is empty
+            $sql = "DROP TABLE `module_prior_authorizations`"; // Drop the module_prior_authorizations table if it is empty
+            $status = sqlStatement($sql); // Execute the SQL statement
+            error_log($logMessage . ' ' . text($status)); // Log the status of the SQL statement
```

Comment:
Need to use errorLogEscape here or use the SystemLogger class which does this for you.